src/Fakta/Agent.fs

```diff
@@ -1,78 +1,389 @@
 ﻿/// Agent can be used to query the Agent endpoints
 module Fakta.Agent
+open System
+open System.Text
+open Fakta
+open Fakta.Logging
+open Fakta.Impl
+open System
+open NodaTime
+open HttpFs.Client
+open Chiron
 
 /// CheckDeregister is used to deregister a check with the local agent
 let checkDeregister (state : FaktaState) (checkId : string) : Async<Choice<unit, Error>> =
-  raise (TBD "TODO")
+  let getResponse = Impl.getResponse state "Fakta.Agent.check.deregister"
+  let req =
+    UriBuilder.ofAgent state.config (sprintf "check/deregister/%s" checkId)
+    |> UriBuilder.uri
+    |> basicRequest Get
+    |> withConfigOpts state.config
+  async {
+  let! resp, dur = Duration.timeAsync (fun () -> getResponse req)
+  match resp with
+  | Choice1Of2 resp ->
+    use resp = resp
+    if not (resp.StatusCode = 200 || resp.StatusCode = 404) then
+      return Choice2Of2 (Message (sprintf "unknown response code %d" resp.StatusCode))
+    else
+      match resp.StatusCode with      
+      | 200 -> return Choice1Of2 ()
+      | _ ->  return Choice2Of2 (Message (sprintf "agent.check.deregister value %s" checkId))
+
+  | Choice2Of2 exx ->
+    return Choice2Of2 (Error.ConnectionFailed exx)
+}
 
 /// CheckRegister is used to register a new check with the local agent 
-let checkRegister (state : FaktaState) (checkId : string) : Async<Choice<unit, Error>> =
-  raise (TBD "TODO")
+let checkRegister (state : FaktaState) (checkRegistration : AgentCheckRegistration) : Async<Choice<unit, Error>> =
+  let getResponse = Impl.getResponse state "Fakta.Agent.check.register"
+  let serializedCheckReg = Json.serialize checkRegistration |> Json.format
+  let req =
+    UriBuilder.ofAgent state.config "check/register"
+    |> UriBuilder.uri
+    |> basicRequest HttpMethod.Put
+    |> withConfigOpts state.config
+    |> withJsonBody serializedCheckReg
+  async {
+  let! resp, dur = Duration.timeAsync (fun () -> getResponse req)
+  match resp with
+  | Choice1Of2 resp ->
+    use resp = resp
+    if not (resp.StatusCode = 200 || resp.StatusCode = 404) then
```

Comment:
This would seem to be repeated throughout the health check? Can you extract to function?